qiskit_ibm_runtime/session.py

```diff
@@ -162,29 +178,23 @@ def run(
 
         options["backend"] = self._backend
 
-        if not self._session_id:
-            # Make sure only one thread can send the session starter job.
-            self._setup_lock.acquire()
+        if self._max_time:
             # TODO: What happens if session max time != first job max time?
             # Use session max time if this is first job.
             options["session_time"] = self._max_time
 
-        try:
-            job = self._service.run(
-                program_id=program_id,
-                options=options,
-                inputs=inputs,
-                session_id=self._session_id,
-                start_session=self._session_id is None,
-                callback=callback,
-                result_decoder=result_decoder,
-            )
-
-            if self._session_id is None:
-                self._session_id = job.job_id()
-        finally:
-            if self._setup_lock.locked():
-                self._setup_lock.release()
+        job = self._service.run(
+            program_id=program_id,
+            options=options,
+            inputs=inputs,
+            session_id=self._session_id,
+            start_session=self._session_id is None,
```

Comment:
I think the plan is to reject jobs that have `start_session=True` once dedicated session is rolled out. I don't know if that applies to simulator jobs, but probably safer to always set this to False.

Reply:
I wouldn't assume that as it's be a breaking api change. POST /jobs should still be possible to create a session. The server side default might change to dedicated mode but that's different.